src/createServer.js

```diff
@@ -1,3 +1,60 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { URL } = require('url')
+const {convertToCase } = require('./convertToCase/convertToCase');
+
+const availableCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+function createServer() {
+  return http.createServer((req, res) => {
+    res.setHeader('Content-type', 'application/json');
+
+    const newUrl = new URL(req.url, `http://${req.headers.host}`);
+
+    const targetCase = newUrl.searchParams.get('toCase');
+    const textToConvert = newUrl.pathname.slice(1);
+
+    const isError = !textToConvert
+    || !targetCase
+    || availableCases.every(availableCase => availableCase !== targetCase);
+
+    if (isError) {
+      const errors = [];
+
+      if (!textToConvert) {
+        errors.push({
+          "message": 'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".'
+        })
+      }
+
+      if (!targetCase) {
+        errors.push({
+          "message": '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+        })
+      }
+
+      if (availableCases.every(availableCase => availableCase !== targetCase && targetCase)) {
```

Comment:
The condition for checking if the targetCase is in availableCases is unnecessary. Just check if the targetCase is not in availableCases.